backends/s3/src/main/scala/geotrellis/s3/S3Client.scala

```diff
@@ -14,10 +14,10 @@ import com.amazonaws.ClientConfiguration
 import org.apache.commons.io.IOUtils
 import scala.collection.JavaConversions._
 
-trait S3Client extends LazyLogging {
+trait S3Client extends LazyLogging with Serializable {
```

Comment:
This seems like a bug. S3Client wraps an object that has references to sockets which are no transferable between machines, even if serializable. This is the reason that s3 reader/writer code will ask for `() => S3Client` instead of `S3Client` directly. The former can be serialized and used to construct a new client on the workers and the latter will cause errors during distributed execution.

Whatever code prompted this change probably has a bug that will reveal itself on a cluster but not on a local test.